### PR TITLE
Stable API tokens in seeds for development

### PR DESCRIPTION
**Pull request by timabell**
### Changes proposed in this pull request

* Stable API tokens in seeds for development
* Remove output from sandbox seeds file, the providers already have their sandbox tokens so this is no longer needed.

### Context

Currently working on https://dfedigital.atlassian.net/browse/CPDTP-232 "Seed API sandbox environment with mentor data"

I wanted to remove the token generation from the sandbox seeds as that has already served its one-time purpose of making tokens for providers to test with and it was getting in the way of the current task.

We have been using that output for local development testing, so in order to continue to facilitate that use of the tokens I've moved it to a more permanent home in the initial_seeds file and made them stable for convenience.

### Guidance to review

Are we happy with well known tokens in the code? I *think* it's a good trade-off of convenience here as it will save a lot of passing around tokens.

### Testing

`bundle exec db:reset`

Try one of the new tokens locally.

### CODE REVIEW

File: db/seeds/initial_seed.rb [hunk 1]
```diff
@@ -59,3 +59,12 @@
...
   NpqCourse.find_or_create_by!(name: hash[:name], id: hash[:id])
 end
+
+if Rails.env.development?
+  LeadProviderApiToken.create_with_known_token!("_488ase4rzqrLo6p3Kzy", lead_provider: ambition)
+  LeadProviderApiToken.create_with_known_token!("KCC_Frtiy79M3zLNMhic", lead_provider: bpn)
+  LeadProviderApiToken.create_with_known_token!("bBvWZ5dr6wmbUNqidnY-", lead_provider: capita)
+  LeadProviderApiToken.create_with_known_token!("kDyukqzjhuPw7j3kgLT2", lead_provider: edt)
+  LeadProviderApiToken.create_with_known_token!("usQncFvBGym5UoGcbsMs", lead_provider: teach_first)
+  LeadProviderApiToken.create_with_known_token!("RCxkriTrzw4yhnn8zjyy", lead_provider: ucl)
```
Comment (ethanmills) [inline db/seeds/initial_seed.rb:69]: Do they even need to be in this format? For dev could they not just be e.g. `ambition-token`?

**Review by ethanmills — APPROVED** [review]

File: db/seeds/initial_seed.rb (hunk 1 above)
Comment (timabell) [reply db/seeds/initial_seed.rb:69]: I guess so, I just kept them as per the generated ones for consistency

File: db/seeds/sandbox_data.rb [hunk 2]
```diff
@@ -14,7 +14,6 @@
 providers_names = ["Capita", "Teach First", "UCL", "Best Practice Network", "Ambition", "Education Development Trust"]
 
 def generate_provider_token(lead_provider, school, cohort, logger)
```
Comment (timabell) [inline db/seeds/sandbox_data.rb:16]: Oops, method name doesn't make sense any more...

File: db/seeds/initial_seed.rb (hunk 1 above)
Comment (timabell) [reply db/seeds/initial_seed.rb:69]: Done, thanks. Also did some refactoring to make it more DRY and make the values that vary stand out.

File: db/seeds/initial_seed.rb [hunk 3]
```diff
@@ -7,29 +7,19 @@
...
+  { provider_name: "Capita", cip: ambition_cip, token: "capita-token" },
+  { provider_name: "Education Development Trust", cip: edt_cip, token: "edt-token" },
+  { provider_name: "Teach First", cip: teach_first_cip, token: "teach-first-token" },
+  { provider_name: "UCL Institute of Education", cip: ucl_cip, token: "ucl-token" },
+].each do |seed|
+  provider = LeadProvider.find_or_create_by!(name: seed[:provider_name])
+  provider.update!(cohorts: [cohort_2021]) unless provider.cohorts.any?
+  LeadProviderCip.find_or_create_by!(lead_provider: provider, cohort: cohort_2021, core_induction_programme: seed[:cip])
+  LeadProviderApiToken.create_with_known_token!(seed[:token], lead_provider: provider)
+end
```
Comment (kicferk1) [inline db/seeds/initial_seed.rb:22]: Soo, making this stuff public is not too bad. But I don't want this in production. Can we hide it behind environment check? Or perhaps move token generation to a seed file that doesn't get run in production?

I know we have already ran the seed there, but, well, better safe than sorry.

Comment (timabell) [reply db/seeds/initial_seed.rb:22]: Damn I forgot to move the environment guard across when I moved it in the refactor commit, good spot @kicferk1 , thank you!

**Review by kicferk1 — APPROVED** [review]

File: db/seeds/initial_seed.rb (hunk 3 above)
Comment (timabell) [reply db/seeds/initial_seed.rb]: Done. :sweat_smile:

**Pull request merged**